### ISSUE TITLE
State in components that are removed are never dropped

### ISSUE THREAD

**Issue by Demonthos** [DioxusLabs/dioxus#769]
**Problem**

The state create in use_state hooks in components that are created and then removed are never have there drop handler called.

**Steps To Reproduce**

Run:
```rust
use dioxus::prelude::*;

fn main() {
    dioxus_desktop::launch(app);
}

fn app(cx: Scope) -> Element {
    let count = if cx.generation() % 2 == 0 { 10 } else { 0 };
    if cx.generation() < 10 {
        cx.needs_update();
    }

    render! {
        (0..count).map(|_| rsx!{
            Comp {}
        })
    }
}

fn Comp(cx: Scope) -> Element {
    cx.use_hook(|| Drops);
    render! {
        div{}
    }
}

struct Drops;

impl Drop for Drops {
    fn drop(&mut self) {
        println!("Dropped!");
    }
}
```

**Expected behavior**

The message Dropped should be called when the program starts, not just when it is closed and the VirtualDom is dropped

**Environment:**
 - Dioxus version: [e.g. v0.17, `master`]
 - Rust version: 1.66.0
 - OS info: Windows
 - App platform: `core`

**Questionnaire**
- [x] #770
- [ ] I would like to fix and I have a solution
- [x] I don't have time to fix this right now, but maybe later

**Comment by jkelleyrtp** [comment]
That not good

**Comment by Demonthos** [comment]
It looks like drop_scope is never called in remove_component here: https://github.com/DioxusLabs/dioxus/blob/master/packages/core/src/diff.rs#L918

**Comment by jkelleyrtp** [comment]
Good catch, yep, the "remove component node" function should just be returning the props from the scope to the vcomponent in case the vcomponent gets rendered again.

**Comment by jkelleyrtp** [comment]
Put up #771

**Comment by Demonthos** [comment]
fixed in #771
